src/main/scala/nl/wbaa/gargoyle/proxy/S3Proxy.scala

```diff
@@ -3,38 +3,35 @@ package nl.wbaa.gargoyle.proxy
 import akka.actor.ActorSystem
 import akka.event.Logging
 import akka.http.scaladsl.Http
+import akka.http.scaladsl.server.directives.DebuggingDirectives
 import akka.stream.ActorMaterializer
+import com.typesafe.config.ConfigFactory
 import com.typesafe.scalalogging.LazyLogging
 import nl.wbaa.gargoyle.proxy.route._
-import akka.http.scaladsl.server.directives.DebuggingDirectives
-import com.typesafe.config.ConfigFactory
 
-import scala.concurrent.Await
-import scala.concurrent.duration.Duration
+import scala.concurrent.Future
 
 class S3Proxy()(implicit system: ActorSystem = ActorSystem.create("gargoyle-s3proxy")) extends LazyLogging {
   import S3Proxy._
 
   implicit val ec = system.dispatcher
-  private var bind: Http.ServerBinding = _
 
-  def start(): Http.ServerBinding = {
+  def start(): Future[Http.ServerBinding] = {
```

Comment:
This is good, but the way callers are using this makes me think it's not behaving as expected…